src/AutoRemoteSyscalls.cc

```diff
@@ -131,6 +132,18 @@ AutoRemoteSyscalls::AutoRemoteSyscalls(Task* t,
     // AutoRemoteSyscall (see below).
     need_sigpending_renable = true;
   }
+  if (t->session().is_recording()) {
+    RecordTask *rt = static_cast<RecordTask*>(t);
+    if (rt->schedule_frozen) {
```

Comment:
Why are we only doing this for `schedule_frozen`? Wouldn't it make sense to do this for all signals?

Reply:
Yes, it makes sense for all tasks, but it does incur two additional syscalls (though if there are actually pending signals, it's much cheaper than what we need to do to handle that case). Usually it's not really a problem to dequeue these signals, we'll just stash them and get back to them later, but in this case, dequeing the signal on the thread is a problem, because we expect the main thread to get to them first.